### DISCUSSION TITLE
Pelican as an OAuth2 provider for external integrations

### DISCUSSION THREAD

**Discussion by TekExplorer** [pelican-dev/panel#445] [Ideas]
I'd like to bring up the possibility of getting pelican to be an oauth2 provider for my mobile app to use 
Originally, this was problematic since I cant exactly create an oauth client on every panel out there, but apparently there exist public clients:
<https://laravel.com/docs/11.x/passport#code-grant-pkce>
- This will require moving from sanctum to passport
- This will also make it so that panel owners can add their own fancy integrations without requiring users to manually make api keys - just run the oauth flow
  - This can be useful for discord bots and other such integrations
  - This will make the panel _much_ more attractive for integrators of all kinds, and i believe that user experience will increase noticeably when it comes up
  - Who knows what tools and apps people will make?
- a "public" client can be created for the purpose of authentication in the app
  - This can allow for the presence of a "devices" page that shows all logged in devices
  - Also has the included benefit of automatically expiring tokens when not used for a while
  - This will allow for any panel to authenticate however they please, including MFA, SSO, and so on, without the app ever needing to be aware of it, which can be good for security and trustworthiness.
 - Activity logs can be more descriptive about what's accessing the api
 - This can be expanded with more granular permissions on both the admin and client sides to have as scopes for applications to use, so that they only request what they need, and no more. (this is, obviously, good for security)
   - My app for example, may only require client scopes, even for admins, and if that ever changes, I can just prompt for a re-authorization with new scopes.
 
This would resolve #120

**Comment by rmartinoscar** [comment]
Thought it was resolved by https://github.com/pelican-dev/panel/issues/385 my bad

**Reply by TekExplorer** [reply]
Is it? That looks more like allowing other oautg providers be used to sign in, not using pelican itself for oauth.
